services/scheduler/src/main/java/org/mule/service/scheduler/internal/DefaultSchedulerService.java

```diff
@@ -97,60 +104,63 @@
 
   @Override
   public String getName() {
-    return "SchedulerService";
+    return SchedulerService.class.getSimpleName();
   }
 
   @Override
   public Scheduler cpuLightScheduler() {
-    checkStarted();
-    final DefaultScheduler scheduler =
-        new DefaultScheduler(resolveSchedulerCreationLocation(CPU_LIGHT_THREADS_NAME), cpuLightExecutor, 4 * cores,
-                             scheduledExecutor, quartzScheduler, CPU_LIGHT, schr -> activeSchedulers.remove(schr));
-    activeSchedulers.add(scheduler);
-    return scheduler;
+    return doCpuLightScheduler(config());
   }
 
   @Override
   public Scheduler ioScheduler() {
-    checkStarted();
-    final DefaultScheduler scheduler = new DefaultScheduler(resolveSchedulerCreationLocation(IO_THREADS_NAME), ioExecutor,
-                                                            cores * cores, scheduledExecutor, quartzScheduler, IO,
-                                                            schr -> activeSchedulers.remove(schr));
-    activeSchedulers.add(scheduler);
-    return scheduler;
+    return doIoScheduler(config());
   }
 
   @Override
   public Scheduler cpuIntensiveScheduler() {
-    checkStarted();
-    final DefaultScheduler scheduler =
-        new DefaultScheduler(resolveSchedulerCreationLocation(COMPUTATION_THREADS_NAME), computationExecutor, 4 * cores,
-                             scheduledExecutor, quartzScheduler, CPU_INTENSIVE, schr -> activeSchedulers.remove(schr));
-    activeSchedulers.add(scheduler);
-    return scheduler;
+    return doCpuIntensiveScheduler(config());
   }
 
   @Override
-  public Scheduler cpuLightScheduler(SchedulerConfig config) {
+  @Inject
+  public Scheduler cpuLightScheduler(@Named(OBJECT_SCHEDULER_BASE_CONFIG) SchedulerConfig config) {
```

Comment:
Can be the injection annotations on the service API instead of in the implementation? Just wondering what would happen if there are different implementations of the service API